### PR TITLE
Expo image

### PR DESCRIPTION
**Pull request by WojtekBoman**
### Details

This PR improves the way of displaying images on mobile platforms. The most relevant changes are the removal of react-native-svg-transformer and the use expo-image to display images in the app. Previously all icons in .svg format were imported as a React components, what in result has negative influence on app's performance. Thanks to the usage of expo-image, we no longer need to transform svg icons and can use them directly by passing them to the source property of the Image component from 'expo-image' library.  

Comparison of the bundle app sizes with and without expo-image:
1. iOS (ReleaseDevelopment)
- With expo-image: 336.6 mb
- Without: 328.9 mb
2. Android (ReleaseProduction)
- With expo-image: 132.8 mb
- Without: 129.1 mb

In case of the iOS Platform were compared compressed archives of development releases. 

### Fixed Issues

Reduced number of rendered elements in the component tree and better rendering times.

<details>
<summary>Rendering SignInPage with react-native-svg on Android</summary>

Screenshot 2023-11-02 at 15 46 52

</details>

<details>
<summary>Rendering SignInPage with expo-image on Android</summary>
Screenshot 2023-11-02 at 15 46 15

</details>

There are not issues associated with this PR.  

$  
PROPOSAL:  https://github.com/Expensify/App/issues/29067#issuecomment-1827768462
Original Problem/Solution was posted here: https://expensify.slack.com/archives/C01GTK53T8Q/p1695892593681869

### Tests

1. Go through the application and check for any problems with displaying images

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Go through the application and check for any problems with displaying images

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/47774969/0cdd19fe-2bfb-4cb3-845a-a8012e9d2830

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/47774969/54135b70-a6e8-492e-aa4f-b0c22b06b621

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/47774969/02d05b49-58ac-40e0-9d7e-8d96c29a8f28

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/47774969/4142caa9-383a-4cce-a44c-94e25466c39b
</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/47774969/04c5f6cb-d285-4636-a25e-74b5bc9e8ac0

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/47774969/2fcdab9c-9f6b-4f98-8c1b-70bba411fdd4

</details>

### CODE REVIEW

**Comment by hayata-suenaga** [comment]
Asked for a C+'s help in reviewing this PR

https://github.com/Expensify/App/pull/30905

**Comment by akinwale** [comment]
@WojtekBoman Could you please fix merge conflicts?

File: src/components/Avatar.js
Comment (akinwale) [inline src/components/Avatar.js:62]: Why is the default fill color set to null here instead of the icon theme color?

File: src/components/BlockingViews/FullPageNotFoundView.js
Comment (akinwale) [inline src/components/BlockingViews/FullPageNotFoundView.js:67]: Instead of passing `undefined` here, I think we should set the default prop value for the `BlockingView` `iconColor` to `undefined`.

File: src/components/DeeplinkWrapper/DeeplinkRedirectLoadingIndicator.js
Comment (akinwale) [inline src/components/DeeplinkWrapper/DeeplinkRedirectLoadingIndicator.js:44]: Here also, set the default value for the `fill` prop in the Icon component to `undefined` instead?

File: src/components/DistanceMapView/index.android.js
Comment (akinwale) [inline src/components/DistanceMapView/index.android.js:34]: Already pointed out. Essentially, if we have the default prop value as undefined, then we can avoid repeatedly setting `iconColor` to undefined when we use the component.

File: src/components/MapView/PendingMapView.tsx
Comment (akinwale) [inline src/components/MapView/PendingMapView.tsx:27]: Pointed out previously. Another case where we can avoid repeatedly setting a prop to undefined when using the component.

File: src/components/RoomHeaderAvatars.js
Comment (akinwale) [inline src/components/RoomHeaderAvatars.js:92]: It looks like the intent is to no longer use a `fill` color for avatars. I already asked above, but it would be helpful to understand why we're doing this.

File: src/components/ValidateCode/ExpiredValidateCodeModal.js
Comment (akinwale) [inline src/components/ValidateCode/ExpiredValidateCodeModal.js:22]: Raised above. Avoidable repetition.

File: src/components/ValidateCode/JustSignedInModal.js
Comment (akinwale) [inline src/components/ValidateCode/JustSignedInModal.js:26]: Also raised above. Avoidable repetition.

File: src/pages/ProfilePage.js
Comment (akinwale) [inline src/pages/ProfilePage.js:269]: Raised above. Avoidable repetition.

File: src/pages/TeachersUnite/ImTeacherUpdateEmailPage.js
Comment (akinwale) [inline src/pages/TeachersUnite/ImTeacherUpdateEmailPage.js:32]: Raised above. Avoidable repetition.

File: src/pages/settings/Security/TwoFactorAuth/Steps/DisabledStep.js
Comment (akinwale) [inline src/pages/settings/Security/TwoFactorAuth/Steps/DisabledStep.js:19]: Raised above. Avoidable repetition.

**Review by akinwale — CHANGES_REQUESTED** [review]
Looks like there are a few instances where we can avoid repetition by making use of default prop values.

Everything else looks good otherwise.

File: src/components/Avatar.js
Comment (WojtekBoman) [reply src/components/Avatar.js:62]: When we set the default value of fill property to `themeColors.icon` it overrides default color of svg icons. It leads to situations when we are unable to display the defaultAvatar component correctly. Passing null value to the fill property indicates that we want to use default fill color of svg icons. The app looks like this when we set the fill property value to `themeColors.icon`

simulator_screenshot_A3E79E43-5243-4C6E-B329-04CA61071F76

File: src/components/BlockingViews/FullPageNotFoundView.js
Comment (WojtekBoman) [reply src/components/BlockingViews/FullPageNotFoundView.js:67]: It's connected with the issue mentioned above. Value undefined indicates that we want to display svg icon in its default color. I've attached the comparison between two blocking views displayed with and without passed undefined value. If we don't pass iconColor={undefined} icon is displayed in `themeColors.offline` color. Solution which can make code more readable is setting default value of iconColor to undefined, and pass `themeColors.offline` to components when it's necessary
Simulator Screenshot - iPhone 15 - 2023-11-14 at 11 18 27
Simulator Screenshot - iPhone 15 - 2023-11-14 at 11 18 54

Comment (akinwale) [reply src/components/BlockingViews/FullPageNotFoundView.js:67]: > Solution which can make code more readable is to set the default value of iconColor to undefined and pass themeColors.offline icon color to components when it's necessary. Will it solve the problem?

Yes, I think we should do this. This would also apply to the other instances I raised in the review. Thanks.

**Comment by akinwale** [comment]
@WojtekBoman Images in the attachment preview are not being displayed on iOS native. See the videos below. Please check.

https://github.com/Expensify/App/assets/4160319/53d0eff7-e24f-4c14-9ac4-1235f39146af

https://github.com/Expensify/App/assets/4160319/2bba91c2-7ad5-4e90-8a29-805c221b81ad

**Comment by akinwale** [comment]
Same bug also happening on Android native.

[Screencast from 15-11-2023 01:42:58.webm](https://github.com/Expensify/App/assets/4160319/51e06ee3-1bd2-48e8-af0f-29d6802a9f33)

**Comment by akinwale** [comment]
@WojtekBoman New error when trying to preview an attachment on native.

Screenshot 2023-11-15 at 14 40 47

**Comment by WojtekBoman** [comment]
> @WojtekBoman New error when trying to preview an attachment on native.
> 
> Screenshot 2023-11-15 at 14 40 47

Yes, I know about it. I haven't finished fixing it yet. I will let you know that PR is ready to review when I finish it

**Comment by WojtekBoman** [comment]
@akinwale I've added some fixes to this PR. Could you please check it again?

**Comment by akinwale** [comment]
@WojtekBoman When we have multiple images in the carousel, the image disappears after trying to navigate back and forth.

https://github.com/Expensify/App/assets/4160319/b406baa9-b6dc-48ff-b743-1c4b2cda0214

**Comment by akinwale** [comment]
@WojtekBoman I ran `npm i` and `pod install` again and it seems to be working fine now. However, when using the arrow buttons to navigate the carousel, the images seem to be flashing instead of sliding. It looks like the adjacent image is unloaded while it is not visible. Video below.

https://github.com/Expensify/App/assets/4160319/c7cfa9b9-cbd0-40a0-96bc-cf5b4d280a63

**Comment by akinwale** [comment]
@WojtekBoman Video of the fast-image implementation for comparison

https://github.com/Expensify/App/assets/4160319/797a18b4-ad93-42d5-a103-c8bda816579b

**Comment by WojtekBoman** [comment]
@akinwale I've added the fix to the AttachmentCarouselPage. Could you please check it?

**Comment by akinwale** [comment]
@WojtekBoman Could you please check this warning which shows up on Android native?

[Screencast from 05-12-2023 07:33:00.webm](https://github.com/Expensify/App/assets/4160319/d817a2dc-8d8c-4b9b-ae73-a5b16cca74ba)

**Comment by WojtekBoman** [comment]
> @WojtekBoman Could you please check this warning which shows up on Android native?
> 
>  Screencast.from.05-12-2023.07.33.00.webm

@akinwale This warning also appears on the main branch. I think that it's connected with the FlashList component in `src/components/LHNOptionsList.js`

**Review by akinwale — APPROVED** [review]
LGTM.

Tests well.

**Comment by WojtekBoman** [comment]
@hayata-suenaga Would you have time to review this?

**Comment by hayata-suenaga** [comment]
wow this is a lot of code changes 😓 

@akinwale thank you for your review. Were you able to check that all images work as expected?

**Review by hayata-suenaga — DISMISSED** [review]
LGTM thank you for the work!

**Comment by hayata-suenaga** [comment]
ops now we have conflicts 😢

**Comment by WojtekBoman** [comment]
Conflicts have been resolved :)

File: jest.config.js
Comment (AndrewGable) [inline jest.config.js:4]: Is this required? If yes, probably should remove the above TODO.

File: src/components/ImageSVG/imageSVGPropTypes.js
Comment (AndrewGable) [inline src/components/ImageSVG/imageSVGPropTypes.js:1]: Why are we adding new prop types instead of writing TS?

Comment (hayata-suenaga) [reply src/components/ImageSVG/imageSVGPropTypes.js:1]: I don't think we have other typescript files ready to implement new component in TypeScript @WojtekBoman for confirmation

File: package.json
Comment (roryabraham) [inline package.json:102]: Do we have to use `expo-asset`? I would prefer to only use `expo-image`. We are [working with Callstack to port react-native-asset directly into the React Native CLI](https://github.com/Expensify/App/issues/13772#issuecomment-1825545287), so that's what we'll prefer to use.

File: src/components/ImageSVG/imageSVGPropTypes.js
Comment (WojtekBoman) [reply src/components/ImageSVG/imageSVGPropTypes.js]: I've migrated ImageSVG component to TS

**Comment by WojtekBoman** [comment]
@roryabraham @AndrewGable Issues mentioned above have been resolved :)

File: android/app/src/main/java/com/expensify/chat/MainActivity.java
Comment (roryabraham) [reply android/app/src/main/java/com/expensify/chat/MainActivity.java:35]: I don't feel strongly or have experience to back this up, my instinct is that departing from the "standard config" for an app including Expo Modules is likely to cause more potential headaches with using upgrades than not.

Comment (AndrewGable) [reply android/app/src/main/java/com/expensify/chat/MainActivity.java:35]: @WojtekBoman or someone else from SWM/Expo- Can you confirm @roryabraham's instincts and/or assumptions above? Thanks.

Comment (WojtekBoman) [reply android/app/src/main/java/com/expensify/chat/MainActivity.java:35]: This is the required configuration to use expo in the project :) More information here: https://docs.expo.dev/bare/installing-expo-modules/

Comment (staszekscp) [reply android/app/src/main/java/com/expensify/chat/MainActivity.java:35]: As far as I know, theoretically it is not required to add those changes, but because that's what the `expo-modules` documentation states I think we should keep it. Especially if we're going to use another module - `expo-av` for video playback.

I think it is also safer to keep those changes, because the expo team would assume that the modules setup is done according to the documentation. It will save us some potential problems during future upgrades.

File: patches/@react-native+assets-registry+0.72.0.patch
Comment (roryabraham) [inline patches/@react-native+assets-registry+0.72.0.patch:1]: Can you please add an explanation and link to the upstream change related to the additional of this patch to the PR description?

File: src/components/BlockingViews/FullPageOfflineBlockingView.js
Comment (roryabraham) [inline src/components/BlockingViews/FullPageOfflineBlockingView.js:8]: `themeColors` should not be used directly. Use the `useTheme` hook instead please

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: patches/@react-native+assets-registry+0.72.0.patch
Comment (WojtekBoman) [reply patches/@react-native+assets-registry+0.72.0.patch]: The issue fixed by this patch no longer occurs, so I've removed this file

File: src/components/SplashScreenHider/index.native.tsx
Comment (roryabraham) [inline src/components/SplashScreenHider/index.native.tsx:5]: NAB but I prefer `SVGImage` over `ImageSVG`

File: src/pages/GetAssistancePage.js
Comment (roryabraham) [inline src/pages/GetAssistancePage.js:54]: I don't see a `name` prop in `MenuItem`?

File: src/pages/signin/SignInPageLayout/BackgroundImage/index.ios.js
Comment (roryabraham) [inline src/pages/signin/SignInPageLayout/BackgroundImage/index.ios.js:23]: Let's update this to:

- Just make the source conditional
- Memoize the source
- Use `StyleUtils.getWidthStyle` instead of an inline style. I know these are basically the same but it's just our code style

```js
const src = useMemo(() => props.isSmallScreen ? MobileBackgroundImage : DesktopBackgroundImage, [props.isSmallScreen]);

return <Image source={src} style={[styles.signInBackground, StyleUtils.getWidthStyle(props.width)]} />
```

**Review by roryabraham — CHANGES_REQUESTED** [review]
Just a few more minor things I noticed

File: src/components/SplashScreenHider/index.native.tsx
Comment (WojtekBoman) [reply src/components/SplashScreenHider/index.native.tsx:5]: You're right` SVGImage` sounds better but there is one benefit of naming this component `ImageSVG`. This allows to store the Image and ImageSVG components side by side in the `src/components` package, which is quite intuitive :)

**Review by roryabraham — DISMISSED** [review]

**Comment by roryabraham** [comment]
Approved, but it looks like we have conflicts again

**Comment by WojtekBoman** [comment]
@roryabraham I've resolved all conflicts :)

**Comment by roryabraham** [comment]
@akinwale there have been a lot of changes since your last approval, can you please re-review and retest this PR?

**Comment by roryabraham** [comment]
Also @WojtekBoman can you look into failing reassure tests? I don't know that I'd block on this as the benchmarks show that this PR results in a great performance improvement, despite a small uptick in the number of renders of the report screen, but if we _can_ get the tests to pass that would be great. Otherwise we'll merge this and establish a new baseline

**Review by akinwale — APPROVED** [review]
LGTM.

Tests well on all platforms.

**Comment by akinwale** [comment]
@roryabraham Just noticed an issue. Please hold off on the merge operation.

**Comment by roryabraham** [comment]
Looks like the issue was reported [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1703193096875019?thread_ts=1703186271.886829&cid=C01GTK53T8Q). Good catch @akinwale

**Comment by WojtekBoman** [comment]
@akinwale Now it should work fine :) Could you please check it again?

**Comment by akinwale** [comment]
Successfully re-tested on all platforms. Everything looks good.

cc @roryabraham

**Review by roryabraham — APPROVED** [review]

**Comment by roryabraham** [comment]
Leaving some additional slack context here: https://expensify.slack.com/archives/C01GTK53T8Q/p1703186271886829

File: package.json
Comment (shubham1206agra) [inline package.json:103]: Is there any specific reason for 1.8.1 not 1.9.0?
cc @mountiny

File: src/components/HeaderWithBackButton/index.js
Comment (VickyStash) [inline src/components/HeaderWithBackButton/index.js:25]: @WojtekBoman, what was the reason for this update? I'm working on the [TS migration](https://github.com/Expensify/App/pull/33214) of this file and want to be sure I haven't missed something.

File: src/components/Image/index.native.js
Comment (shubham1206agra) [inline src/components/Image/index.native.js:37]: Comment for reference

**Comment by roryabraham** [comment]
Looks like it was a bit rocky and we have a bit of cleanup to do, but this is on prod now

File: src/components/Icon/index.tsx
Comment (c3024) [inline src/components/Icon/index.tsx:81]: This PR caused a regression in ReceiptEmptyState.js #33591

**Comment by rushatgabhane** [comment]
This PR caused a tiny regression. `EnableStep` component was missed during refactor.

image

File: src/components/AddPlaidBankAccount.js
Comment (shubham1206agra) [inline src/components/AddPlaidBankAccount.js:252]: This created a regression due to expo https://github.com/Expensify/App/issues/33664

**Comment by jjcoffee** [comment]
Coming from https://github.com/Expensify/App/issues/34907. Similar regression as the one mentioned [here](https://github.com/Expensify/App/pull/30905#discussion_r1463416802), but for EReceipt icons. Expo seems to have an issue with getting the currentColor prop on Android, so the icons were rendering without the correct colour.

**Pull request merged**
